File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {

        String answer = "";
        // 'a' 보다 크면 무조건 알파벳
        for(int i = 0; i < s.length();){
            char c = s.charAt(i);
            if(c >= 'e'){
                if(c >= 'f'){
                    if(c >= 'n'){
                        if(c >= 'o'){
                            if(c >= 's'){
                                if(c >= 't'){
                                    // 0
                                    if(c == 'z') {
                                        answer += "0";
                                        i += 4;
                                        continue;
                                    }
                                    // two or three
                                    else{
                                        if(s.charAt(i+1) == 'w'){
                                            answer += "2";
                                            i += 3;
                                            continue;
                                        }
                                        else{
                                            answer += "3";
                                            i += 5;
                                            continue;
                                        }
                                    }
                                }
                                // six or seven
                                else{
                                    if(s.charAt(i+1) == 'i'){
                                        answer += "6";
                                        i += 3;
                                        continue;
                                    }
                                    else{
                                        answer += "7";
                                        i += 5;
                                        continue;
                                    }
                                    
                                }
                                
                            }
                            // 무조건 1
                            else{
                                answer += "1";
                                i += 3;
                                continue;
                            }
                        }
                        // 무조건 nine
                        else{
                            answer += "9";
                            i += 4;
                            continue;
                        }
                        
                    }
                    // five or four
                    else{
                        if(s.charAt(i+1) == 'o'){
                            answer += "4";
                        }
                        else{
                            answer += "5";
                        }
                        i += 4;
                        continue;
                    }
                }
                // 무조건 eight
                else{
                    answer += "8";
                    i += 5;
                    continue;
                }
            }
            else{
                answer += String.valueOf(s.charAt(i));
                i++;
            }
        }
        
        return Integer.parseInt(answer);
    }
}

File: Programmers/Yoo-Seung-Hui/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    # answer str으로 선언
    answer = ''
    
    # eng_list 선언
    eng_list = ['zero','one','two','three',
               'four','five','six','seven',
               'eight','nine']
    
    # 영어로 되어 있는거 숫자로 반환할 str 선언
    engton = ''
    
    # 문자열 앞에서 부터
    for i in s :
        
        # 알파벳이면
        if i.isalpha():
            
            # 영어 문자열에 추가해주고
            engton += i
            
            # 해당 영문자가 eng_list 에 포함되어 있으면 
            if engton in eng_list :
                
                # 해당 영문자의 인덱스를 문자열로 추가해주기
                answer += str(eng_list.index(engton))
                # 다음 영문자를 숫자로 변환하기 위해 비워주기
                engton = ''
                
        # 알파벳 아니고 숫자면
        else : 
            # 그냥 추가
            answer += i
            
        
    return int(answer)